ui/splash/build.gradle.kts

```diff
@@ -0,0 +1,81 @@
+plugins {
+    id("kotlin-multiplatform")
+    id("org.jetbrains.compose")
+    id("com.android.library")
+    id("com.diffplug.spotless") version "6.11.0"
+    id("kotlin-parcelize")
+}
+
+spotless {
+    kotlin {
+        target("src/*/kotlin/**/*.kt")
+        ktlint("0.43.2")
+        licenseHeaderFile(rootProject.file("copyright.kt"))
+    }
+}
+
+val targetSDKVersion: Int by rootProject.extra
+val minSDKVersion: Int by rootProject.extra
+val compileSDKVersion: Int by rootProject.extra
+
+android {
+
+    namespace = "social.androiddev.ui.splash"
+
+    compileSdk = compileSDKVersion
+
+    defaultConfig {
+        minSdk = minSDKVersion
+        targetSdk = targetSDKVersion
+    }
+
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
+    }
+
+    sourceSets {
+        named("main") {
+            manifest.srcFile("src/androidMain/AndroidManifest.xml")
+            res.srcDirs("src/androidMain/res")
+        }
+    }
+}
+
+kotlin {
+    jvm("desktop")
+    android()
+
+    sourceSets {
+        val commonMain by getting {
+            dependencies {
+                implementation(projects.ui.common)
```

Comment:
Splash UI module depends on other UI modules, I think is not tolerated by the project architecture to have a dependency between UI modules, Or the Splash module is an exception because it contains the `RootComponent`?

Reply:
So `ui:common` can be depended on by other `ui` modules. We can update that doc so it's clear. As for `welcome` and `timeline, maybe the root component should expose callbacks for "welcome" (sign-up/login) and "timeline" (logged in) flows, so that the app modules then route to the respective feature modules where they contain their own navigation graph

What is tricky is the "component"s in decompose contain the navigation logic, and the initial splash screen is the true root of everything, deciding if the user is logged in or not and where to go next. I'm open to suggestions on some tweaks I can make here, but also I'm in favor of moving some things along that aren't "perfect" just to be able to get some more stuff in place quickly, so folks can find more opportunities to jump in and contribute.

Reply:
Compose encourages tree-structures for views, so it's convenient to organize feature modules into trees as well. Personally I don't see any issues with this, and I actually prefer tree-structured feature modules.

Here is an example of possible structure (based on Twitter app). App modules depend on the top `Root` module.
<img width="512" src="https://user-images.githubusercontent.com/26204457/204907146-0dbad51d-ccd0-44a5-bb18-095dd669a614.png">

Reply:
Here's an updated project structure to reflect the changes made in this PR. 

<img width="901" alt="Screen Shot 2022-12-02 at 12 23 01 AM" src="https://user-images.githubusercontent.com/6646320/205220819-7782fc0e-bea5-416c-8813-c5e41c89124d.png">